### DISCUSSION TITLE
dehydratedState not working on linux

### DISCUSSION THREAD

**Discussion by hamsadeh** [TanStack/query#4176] [Q&A]
### Describe the bug

I'm using react-query and next.js to have SSR (using dehydration). I return the dehydrated cache with the dehydratedState prop from the getServerSideProps method, and it works fine on my localhost (Windows). But when I run the same exact code on a Linux server, the dehydratedState prop is empty. 

Here is my _app.js:
```
import React from 'react';

import { QueryClient, QueryClientProvider, Hydrate } from '@tanstack/react-query';

function MyApp({ Component, pageProps }) {

  const queryClient = new QueryClient();
  
  return (

    <QueryClientProvider client={queryClient}>

      <Hydrate state={pageProps.dehydratedState}>
            <Component {...pageProps} />
      </Hydrate>

    </QueryClientProvider>

  )
}
export default MyApp
```

index.js:

```
import { dehydrate, QueryClient } from '@tanstack/react-query';
import titleService from '../src/services/titleService';

export default function Index(props) {

  console.log(props);

  return null
}


export async function getServerSideProps(context){

  const q = context.query['q'];
  
  const queryClient = new QueryClient();
  const {searchTitles} = titleService();
  await queryClient.prefetchQuery(['searchTitles', q], ()=>searchTitles(q));

  return { props: { dehydratedState: dehydrate(queryClient)}};

}
```

### Your minimal, reproducible example

N/A

### Steps to reproduce

1.Just take a look at the code

### Expected behavior

The dehydratedState prop has an array of queries on my localhost (Windows):
{mutations: Array(0), queries: Array(1)}

But it is empty on my Linux Server:
{mutations: Array(0), queries: Array(0)}

### How often does this bug happen?

Every time

### Platform

OS: Linux

### react-query version

3.39.2

### Additional context

 "@tanstack/react-query": "^4.3.4"
"react": "17.0.2"
"next": "12.2.5"

**Comment by TkDodo** [comment]
I would check if `searchTitles` works, or if the query fails. try stubbing it out with a `Promise.resolve(someData)`. We've recently had something similar where server-to-server communication wasn't working due to some SSL issues. Certainly not a bug in react-query. Will move to discussion.
